httpserver/http_connection.cpp

```diff
@@ -546,6 +543,26 @@ bool connection_t::parse_header(const char* first, Args... args) {
     return parse_header(first) && parse_header(args...);
 }
 
+json connection_t::get_snodes_json_by_pk(const std::string& pubKey) const {
```

Comment:
Should this be a freestanding function that accepts `std::vector<sn_record_t>` and returns a json object? I think it is better to keep the API lean.

Reply:
V tru